ffs/scheduler/scheduler.go

```diff
@@ -163,6 +175,50 @@ func (s *Scheduler) Cancel(jid ffs.JobID) error {
 	return nil
 }
 
+// GCStaged runs a unpinned garbage collection of stage-pins.
+func (s *Scheduler) GCStaged(ctx context.Context) ([]cid.Cid, error) {
+	return s.gcStaged(ctx, 0)
+}
+
+// PinnedCids returns the pinned cids from Hot-Storage.
+func (s *Scheduler) PinnedCids(ctx context.Context) ([]ffs.PinnedCid, error) {
+	res, err := s.hs.PinnedCids(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("getting pinned cids from hot-storage: %s", err)
+	}
+	return res, nil
+}
+
+func (s *Scheduler) gcStaged(ctx context.Context, gracePeriod time.Duration) ([]cid.Cid, error) {
+	s.gcLock.Lock()
+	defer s.gcLock.Unlock()
+	log.Infof("running scheduler gc...")
+	cids := map[cid.Cid]struct{}{}
+
+	qj := s.sjs.QueuedJobs(ffs.EmptyInstanceID)
+	for _, j := range qj {
+		cids[j.Cid] = struct{}{}
+	}
+	ej := s.sjs.ExecutingJobs(ffs.EmptyInstanceID)
+	for _, j := range ej {
+		cids[j.Cid] = struct{}{}
+	}
+
+	excludedCids := make([]cid.Cid, 0, len(cids))
+	for c := range cids {
+		excludedCids = append(excludedCids, c)
+	}
+
+	gced, err := s.hs.GCStaged(ctx, excludedCids, time.Now().Add(-gracePeriod))
+	if err != nil {
+		return nil, fmt.Errorf("hot-storage gc: %s", err)
+	}
```

Comment:
We ask Hot-Storage to GC all Cids that *only* have staged-pins, aren't in the exclusion list, and were staged-pinned before the grace-period.